apps/cowswap-frontend/src/modules/tokens/hooks/useEnoughBalance.ts

```diff
@@ -32,50 +34,54 @@ export interface UseEnoughBalanceParams {
  * @returns true if the account has enough balance (and allowance if it applies)
  */
 export function useEnoughBalanceAndAllowance(params: UseEnoughBalanceParams): boolean | undefined {
-  const { account, amount, checkAllowanceAddress } = params
+  const { account, amount, spender } = params
   const isNativeCurrency = amount?.currency.isNative
   const token = amount?.currency.wrapped
 
   const { balances, allowances } = useBalancesAndAllowances({
     account,
-    spender: checkAllowanceAddress,
+    spender,
     tokens: !isNativeCurrency && account && token ? [token as Token] : [],
   })
 
   const native = useNativeCurrency()
   const [nativeBalance] = useCurrencyBalances(isNativeCurrency ? account : undefined, [native])
 
-  return hasEnoughBalanceAndAllowance({
-    ...params,
-    balances,
-    allowances: checkAllowanceAddress ? allowances : undefined,
-    nativeBalance,
-  })
+  return useMemo(() => {
+    return hasEnoughBalanceAndAllowance({
+      account,
+      amount,
+      balances,
+      spender,
+      allowances: spender ? allowances : undefined,
+      nativeBalance,
+    })
+  }, [account, amount, balances, allowances, nativeBalance, spender])
 }
 
-export interface EnoughBalanceParams extends Omit<UseEnoughBalanceParams, 'checkAllowanceAddress'> {
+export interface EnoughBalanceAndAllowanceParams extends UseEnoughBalanceParams {
   /**
    * Balances per token for the account
    */
   balances: TokenAmounts
 
   /**
-   * Allowances per token for the account
+   * Allowances per token for the account. Undefined if the allowances are not required for the check, so only the balance is checked.
    */
-  allowances?: TokenAmounts
+  allowances: TokenAmounts | undefined
```

Comment:
Just tried to make it more explicit that undefined is how you signal you don't want to check the allowances